orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -0,0 +1,314 @@
+package orderbookusecase_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	cltypes "github.com/osmosis-labs/osmosis/v25/x/concentrated-liquidity/types"
+	"github.com/osmosis-labs/sqs/domain/mocks"
+	orderbookdomain "github.com/osmosis-labs/sqs/domain/orderbook"
+	orderbookusecase "github.com/osmosis-labs/sqs/orderbook/usecase"
+	"github.com/osmosis-labs/sqs/sqsdomain/cosmwasmpool"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/osmosis/v25/app/apptesting"
+)
+
+type OrderbookUsecaseTestSuite struct {
+	apptesting.ConcentratedKeeperTestHelper
+}
+
+func TestOrderbookUsecaseTestSuite(t *testing.T) {
+	suite.Run(t, new(OrderbookUsecaseTestSuite))
+}
+
+func (s *OrderbookUsecaseTestSuite) TestCreateFormattedLimitOrder() {
+	testCases := []struct {
+		name          string
+		order         orderbookdomain.Order
+		quoteAsset    orderbookdomain.Asset
+		baseAsset     orderbookdomain.Asset
+		setupMocks    func(orderbookRepo *mocks.OrderbookRepositoryMock, tokensUsecase *mocks.TokensUsecaseMock)
+		expectedError string
+		expectedOrder orderbookdomain.LimitOrder
+	}{
+		{
+			name: "tick not found",
+			order: orderbookdomain.Order{
+				TickId: 99, // Non-existent tick ID
+			},
+			expectedError: "tick not found",
+			setupMocks: func(orderbookRepo *mocks.OrderbookRepositoryMock, tokensUsecase *mocks.TokensUsecaseMock) {
+				orderbookRepo.GetTickByIDFunc = func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
+					return orderbookdomain.OrderbookTick{
+						Tick: &cosmwasmpool.OrderbookTick{
+							TickId: tickID,
+						},
+					}, false
+				}
+			},
+		},
+		{
+			name: "error parsing quantity",
+			order: orderbookdomain.Order{
+				Quantity: "invalid", // Invalid quantity
+			},
+			expectedError: "error parsing quantity",
+			setupMocks: func(orderbookRepo *mocks.OrderbookRepositoryMock, tokensUsecase *mocks.TokensUsecaseMock) {
+				orderbookRepo.GetTickByIDFunc = func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
+					return orderbookdomain.OrderbookTick{
+						Tick: &cosmwasmpool.OrderbookTick{
+							TickId: tickID,
+						},
+					}, true
+				}
+			},
+		},
+		{
+			name: "error parsing placed quantity",
+			order: orderbookdomain.Order{
+				Quantity:       "1000",
+				PlacedQuantity: "invalid", // Invalid placed quantity
+			},
+			expectedError: "error parsing placed quantity",
+			setupMocks: func(orderbookRepo *mocks.OrderbookRepositoryMock, tokensUsecase *mocks.TokensUsecaseMock) {
+				orderbookRepo.GetTickByIDFunc = func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
+					return orderbookdomain.OrderbookTick{
+						Tick: &cosmwasmpool.OrderbookTick{
+							TickId: tickID,
+						},
+					}, true
+				}
+			},
+		},
+		{
+			name: "error getting spot price scaling factor",
+			order: orderbookdomain.Order{
+				Quantity:       "1000",
+				PlacedQuantity: "1500",
```

Comment:
The main reason I am not reusing contants here was to make values local to the test case to avoid adding cognitive load. Let me know if that makes sense.